src/cdk/stepper/stepper.ts

```diff
@@ -117,18 +159,38 @@ export class CdkStep implements OnChanges {
   /** Whether step is marked as completed. */
   @Input()
   get completed(): boolean {
-    return this._customCompleted == null ? this._defaultCompleted() : this._customCompleted;
+    return this._customCompleted == null ? this._getDefaultCompleted() : this._customCompleted;
   }
   set completed(value: boolean) {
     this._customCompleted = coerceBooleanProperty(value);
   }
   private _customCompleted: boolean | null = null;
 
-  private _defaultCompleted() {
+  private _getDefaultCompleted() {
     return this.stepControl ? this.stepControl.valid && this.interacted : this.interacted;
   }
 
-  constructor(@Inject(forwardRef(() => CdkStepper)) private _stepper: CdkStepper) { }
+  /** Whether step has an error. */
+  @Input()
+  get hasError(): boolean {
+    return this._customError || this._getDefaultError();
+  }
+  set hasError(value: boolean) {
+    this._customError = coerceBooleanProperty(value);
+  }
+  private _customError: boolean | null = null;
+
+  private _getDefaultError() {
+    return this.stepControl && this.stepControl.invalid;
+  }
+
+  constructor(
+    @Inject(forwardRef(() => CdkStepper)) private _stepper: CdkStepper,
+    @Optional() @Inject(MAT_STEPPER_GLOBAL_OPTIONS) stepperOptions: StepperOptions) {
```

Comment:
@arodr967 This breaks some teams at Google because they're extending the stepper. We should make the `stepperOptions` parameter optional (`?` in addition to `@Optional()`). That way if they don't pass it in their code will still compile. You can leave a comment to make it mandatory again in the future:

```ts
/** @breaking-change 8.0.0 remove the `?` after `stepperOptions` */
```

Reply:
Ok perfect, thanks for the heads up!